cmd/collector/app/zipkin/http_handler.go

```diff
@@ -75,30 +75,36 @@ func (aH *APIHandler) saveSpans(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if r.Header.Get("Content-Type") == "application/x-thrift" {
-		handleZipkinThrift(aH.zipkinSpansHandler, bodyBytes, w)
+	contentType := r.Header.Get("Content-Type")
+	var tSpans []*zipkincore.Span
+	if contentType == "application/x-thrift" {
+		tSpans, err = deserializeThrift(bodyBytes)
+		if err != nil {
+			http.Error(w, fmt.Sprintf(app.UnableToReadBodyErrFormat, err), http.StatusBadRequest)
+			return
+		}
+	} else if contentType == "application/json" {
+		tSpans, err = deserializeJSON(bodyBytes)
+		if err != nil {
+			http.Error(w, fmt.Sprintf(app.UnableToReadBodyErrFormat, err), http.StatusBadRequest)
+			return
+		}
 	} else {
 		http.Error(w, "Only Content-Type:application/x-thrift is supported at the moment", http.StatusBadRequest)
```

Comment:
I meant err msg says only x-thrift is supported, not json, so we can extend it to mention both